src/coreclr/vm/peimagelayout.cpp

```diff
@@ -616,17 +616,26 @@ FlatImageLayout::FlatImageLayout(PEImage* pOwner)
         PRECONDITION(CheckPointer(pOwner));
     }
     CONTRACTL_END;
-    m_pOwner=pOwner;
-
-    HANDLE hFile = pOwner->GetFileHandle();
-    INT64 offset = pOwner->GetOffset();
-    INT64 size = pOwner->GetSize();
+    m_pOwner = pOwner;
 
 #ifdef LOGGING
     SString ownerPath{ pOwner->GetPath() };
     LOG((LF_LOADER, LL_INFO100, "PEImage: Opening flat %s\n", ownerPath.GetUTF8()));
 #endif // LOGGING
 
+// TODO: Any image from external_assembly_probe should go down this path, not just on Android
+#if defined(TARGET_ANDROID)
```

Comment:
I believe that the existing Android tests in dotnet/runtime expect (unbundled) files on disk. Do we need to turn these ifdefs into runtime conditions to keep these tests working?

Reply:
Switched to check for non-file image with non-null data.